src/tools/rbd/action/Import.cc

```diff
@@ -132,6 +138,30 @@ static int do_import(librbd::RBD &rbd, librados::IoCtx& io_ctx,
 #endif
   }
 
+  if (import_snaps || !from_stdin) {
+    r = safe_read_exact(fd, image_buf, utils::RBD_IMAGE_BANNER.size());
+    if (r < 0)
```

Comment:
Aha, yes, good catch, actually I noticed and fixed this problem in my local tree yesterday. :) thanx.